src/fights.ts

```diff
@@ -351,6 +358,26 @@ function pygmyOptions(forceEquip: Item[] = []) {
   };
 }
 
+let _bestShadowRift: Location | null = null;
+export function bestShadowRift(): Location {
+  if (!_bestShadowRift) {
+    _bestShadowRift =
+      chooseRift({
+        canAdventure: true,
+        sortBy: (l: Location) => {
+          const drops = getMonsters(l)
+            .map((m) => Object.keys(itemDrops(m)).map((s) => toItem(s)))
```

Comment:
Not all drops drop at the same rate, and we probably aren't capping them, so it might be worth using an `Object.entries` to weight them by drop rate